user_modules/utils.lua

```diff
@@ -1,111 +1,144 @@
-  if not interface_schema.enum[enum_name] then
-    common_functions:UserPrint(31, "Enum with name:", " ")
-    common_functions:UserPrint(0, enum_name, " ")
-    common_functions:UserPrint(31, "does not exist")
-    return nil
-  end
-  return #module.GetUnorderedTableKeyset(interface_schema.enum[enum_name])
+--[[ @getDeviceName: provide device name
+--! @parameters: none
+--! @return: name of the device
+--]]
+function m.getDeviceName()
+  return config.mobileHost .. ":" .. config.mobilePort
 end
 
---! @brief Function allows to get value from any mobile api function
---! @param function_type - request, response or notification
---! @param function_name - name of the function
---! @param param_name - function parameter
---! @param value_to_read - value which is needed to be read
---! @param Function usage example: maxlength = module.GetFunctionValueFromMobileApi("request", "Show", "mainField2", "maxlength")
-function module.GetFunctionValueFromMobileApi(function_type, function_name, param_name, value_to_read)
-  if not interface_schema.type[function_type] then
-    common_functions:UserPrint(31, "Function with type:", " ")
-    common_functions:UserPrint(0, function_type, " ")
-    common_functions:UserPrint(31, "does not exist")
-    return nil
-  end
-  if not interface_schema.type[function_type].functions[function_name] then
-    common_functions:UserPrint(31, "Function with name:", " ")
-    common_functions:UserPrint(0, function_name, " ")
-    common_functions:UserPrint(31, "does not exist")
-    return nil
-  end
-  if not interface_schema.type[function_type].functions[function_name].param[param_name] then
-    common_functions:UserPrint(31, "Parameter with name:", " ")
-    common_functions:UserPrint(0, param_name, " ")
-    common_functions:UserPrint(31, "does not exist")
-    return nil
-  end
-  return interface_schema.type[function_type].functions[function_name].param[param_name][value_to_read]
+--[[ @getDeviceMAC: provide device MAC address
+--! @parameters: none
+--! @return: MAC address of the device
+--]]
+function m.getDeviceMAC()
+  local cmd = "echo -n " .. m.getDeviceName() .. " | sha256sum | awk '{printf $1}'"
+  local handle = io.popen(cmd)
+  local result = handle:read("*a")
+  handle:close()
+  return result
 end
 
-return module
+--[[ @protect: make table immutable
+--! @parameters:
+--! pTbl - mutable table
+--! @return: immutable table
+--]]
+function m.protect(pTbl)
+  local mt = {
+    __index = pTbl,
+    __newindex = function(_, k, v)
+      error("Attempting to change item " .. tostring(k) .. " to " .. tostring(v), 2)
+    end
+  }
+  return setmetatable({}, mt)
+end
+
+--[[ @inheritObjects: copy objects from source table to target
+-- 'objects' means: tables, functions, fields
+-- Function useful for 'inheriting' data of one module to another
+--! @parameters:
+--! pTbl - mutable table
+--! @return: immutable table
+--]]
+function m.inheritObjects(pSourceObject, pTargetObject)
```

Comment:
@dboltovskyi I would propose change signature of function `inheritObjects()` switching parameters:
```
function m.inheritObjects(pSourceObject, pTargetObject) 
=> 
function m.inheritObjects(pTargetObject, pSourceObject)
```
The reason for this change is simplifying of understanding of function:
Call of `m.inheritObjects(pTargetObject, pSourceObject)` means `pTargetObject` `inheritObjects` from `pSourceObject`.

Reply:
@aderiabin Accepted. Done in 9a1a42b